src/transformers/models/imagegpt/image_processing_imagegpt_fast.py

```diff
@@ -0,0 +1,251 @@
+            - `TensorType.JAX` or `'jax'`: Return a batch of type `jax.numpy.ndarray`.
+    data_format (`ChannelDimension` or `str`, *optional*):
+        The channel dimension format for the output image. If unset, the channel dimension format of the input
+        image is used. Can be one of:
+            - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+            - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+            - `"none"` or `ChannelDimension.NONE`: image in (height, width) format.
+    input_data_format (`ChannelDimension` or `str`, *optional*):
+        The channel dimension format for the input image. If unset, the channel dimension format is inferred
+        from the input image. Can be one of:
+            - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+            - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+            - `"none"` or `ChannelDimension.NONE`: image in (height, width) format.
+    do_color_quantize (`bool`, *optional*, defaults to `self.do_color_quantize`):
+        Whether to color quantize the image.
+    """
+
+    clusters: Optional[np.ndarray] = None
+    resample: Optional[PILImageResampling] = PILImageResampling.BILINEAR
+    return_tensors: Optional[Union[str, TensorType]] = (None,)
+    data_format: Optional[Union[str, ChannelDimension]] = (ChannelDimension.FIRST,)
+    input_data_format: Optional[Union[str, ChannelDimension]] = None
+    do_color_quantize: Optional[bool] = True
+
+
+@auto_docstring
+class ImageGPTImageProcessorFast(BaseImageProcessorFast):
+    """
+    Constructs a fast ImageGPT image processor.
+
+    This processor can be used to resize images to a smaller resolution (such as 32x32 or 64x64),
+    normalize them and finally color quantize them to obtain sequences of "pixel values" (color clusters).
+    """
+
+    model_input_names = ["input_ids"]
+
+    # Defaults largely aligned with the slow processor, except normalization which we do manually to [-1, 1]
+    resample = PILImageResampling.BILINEAR
+    size = {"height": 256, "width": 256}
+    do_resize = True
+    # We do NOT want touse the base normalization/rescale as ImageGPT expects (x/127.5 - 1)
+    do_rescale = False
+    do_normalize = False
+
+    do_color_quantize = True
+    clusters = None
+
+    def __init__(
+        self,
+        clusters: Optional[Union[list, np.ndarray]] = None,
+        **kwargs: Unpack[ImageGPTFastImageProcessorKwargs],
+    ):
+        super().__init__(**kwargs)
+        # Store clusters as numpy for JSON serializability. Convert to torch in _preprocess when needed.
+        if clusters is not None:
+            self.clusters = np.asarray(clusters, dtype=np.float32)
+        else:
+            self.clusters = None
```

Comment:
why not define as torch tensor directly and convert from torch to list in to_dict?
```suggestion
        self.clusters = torch.tensor(clusters, dtype=torch.float32) if clusters is not None else None
```